File: DynamicKey/AgoraDynamicKey/java/src/main/java/io/agora/sample/ChatTokenBuilder2Sample.java
package io.agora.sample;

import io.agora.chat.ChatTokenBuilder2;

public class ChatTokenBuilder2Sample {
    // Need to set environment variable AGORA_APP_ID
    private static String appId = System.getenv("AGORA_APP_ID");
    // Need to set environment variable AGORA_APP_CERTIFICATE
    private static String appCertificate = System.getenv("AGORA_APP_CERTIFICATE");

    private static String userId = "2882341273";
    private static int expire = 600;

    public static void main(String[] args) throws Exception {
        System.out.printf("App Id: %s\n", appId);
        System.out.printf("App Certificate: %s\n", appCertificate);
        if (appId == null || appId.isEmpty() || appCertificate == null || appCertificate.isEmpty()) {
            System.out.printf("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE\n");
            return;
        }

        ChatTokenBuilder2 tokenBuilder = new ChatTokenBuilder2();

        String userToken = tokenBuilder.buildUserToken(appId, appCertificate, userId, expire);
        System.out.printf("Chat user token: %s\n", userToken);

        String appToken = tokenBuilder.buildAppToken(appId, appCertificate, expire);
        System.out.printf("Chat app token: %s\n", appToken);
    }
}

File: DynamicKey/AgoraDynamicKey/kotlin/src/sample/kotlin/io/agora/sample/ChatTokenBuilder2Sample.kt
package io.agora.sample

import io.agora.media.ChatTokenBuilder2

/** Demonstrates Chat Token007 generation. */
object ChatTokenBuilder2Sample {
    /** Generates and prints Chat user and application sample tokens. */
    @JvmStatic
    fun main(args: Array<String>) {
        val appId = System.getenv("AGORA_APP_ID")
        val appCertificate = System.getenv("AGORA_APP_CERTIFICATE")
        if (appId.isNullOrEmpty() || appCertificate.isNullOrEmpty()) {
            println("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE")
            return
        }

        val builder = ChatTokenBuilder2()
        println("Chat user token: ${builder.buildUserToken(appId, appCertificate, USER_ID, EXPIRE)}")
        println("Chat app token: ${builder.buildAppToken(appId, appCertificate, EXPIRE)}")
    }

    private const val USER_ID = "2882341273"
    private const val EXPIRE = 600
}
